lint-tasks.yml

```
@@ -0,0 +1,217 @@
+
+  fix:
+    cmds:
+      - task: "cmake-fix"
+      - task: "cpp-fix"
+      - task: "py-fix"
+      - task: "yml-fix"
+
+  cmake-check:
+    deps: ["venv"]
+    cmds:
+      - task: "cmake"
+        vars:
+          FLAGS: "--check"
+
+  cmake-fix:
+    deps: ["venv"]
+    cmds:
+      - task: "cmake"
+        vars:
+          FLAGS: "--in-place"
+
+  cpp-configs:
+    # TODO: remove this once all clang-tidy has been fixed
+    deps: [":config-cmake-project"]
+    cmds:
+      - "tools/yscope-dev-utils/lint-configs/symlink-cpp-lint-configs.sh"
+
+  cpp-check:
+    cmds:
+      - task: "cpp-format-check"
+      # TODO: re-enable this once all clang-tidy has been fixed
+      # - task: "cpp-static-check"
+
+  cpp-fix:
+    cmds:
+      - task: "cpp-format-fix"
+      # TODO: re-enable this once all clang-tidy has been fixed
+      # - task: "cpp-static-fix"
+
+  cpp-format-check:
+    sources: &cpp_format_src_files
+      - "{{.LINT_VENV_CHECKSUM_FILE}}"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.cpp"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.h"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.hpp"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.inc"
+      - "src/.clang-format"
+      - "{{.TASKFILE}}"
+      - ".clang-format"
+    deps: ["cpp-configs", "venv"]
+    cmds:
+      - task: ":utils:clang-format"
+        vars:
+          FLAGS: "--dry-run"
+          SRC_PATHS:
+            ref: ".CPP_LINT_DIRS"
+          VENV_DIR: "{{.LINT_VENV_DIR}}"
+
+  cpp-format-fix:
+    sources: *cpp_format_src_files
+    deps: ["cpp-configs", "venv"]
+    cmds:
+      - task: ":utils:clang-format"
+        vars:
+          FLAGS: "-i"
+          SRC_PATHS:
+            ref: ".CPP_LINT_DIRS"
+          VENV_DIR: "{{.LINT_VENV_DIR}}"
+
+  cpp-static-check:
+    # Alias task to `cpp-static-fix` since we don't currently support automatic fixes.
+    # NOTE: clang-tidy does have the ability to fix some errors, but the fixes can be inaccurate.
+    # When we eventually determine which errors can be safely fixed, we'll allow clang-tidy to
+    # fix them.
+    aliases: ["cpp-static-fix"]
+    sources:
+      - "{{.LINT_VENV_CHECKSUM_FILE}}"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.cpp"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.h"
+      - "{{.CLP_FFI_PY_SRC_DIR}}/**/*.hpp"
+      - "{{.CLP_FFI_PY_CMAKE_CACHE}}"
+      - "{{.CLP_FFI_PY_COMPILE_COMMANDS_DB}}"
+      - "{{.TASKFILE}}"
+      - "Taskfile.yml"
+      - ".clang-tidy"
+    deps: [":config-cmake-project", "cpp-configs", "venv"]
+    cmds:
+      - task: ":utils:clang-tidy"
+        vars:
+          FLAGS: "--config-file=.clang-tidy -p {{.CLP_FFI_PY_COMPILE_COMMANDS_DB}}"
+          SRC_PATHS:
+            ref: ".CPP_LINT_DIRS"
+          VENV_DIR: "{{.LINT_VENV_DIR}}"
+
+  py-check:
+    cmds:
+      - task: "mypy"
+        vars:
+          SRC_DIRS:
+            ref: ".PYTHON_LINT_DIRS"
+      - task: "py"
+        vars:
+          DOCFORMATTER_FLAGS: "--check --diff"
+          BLACK_FLAGS: "--check"
+          RUFF_FLAGS: ""
```

```suggestion
          RUFF_FLAGS: "--output-format=github"
```
How about adding this to be consistent with our current workflow check?

---

I got 
`Usage: black [OPTIONS] SRC ...
Try 'black -h' for help.
Error: No such option: --output-format`

what's the version of black that you are using?